src/SFA.DAS.EmployerAccounts.ReadStore/SFA.DAS.EmployerAccounts.ReadStore/Application/Commands/CreateAccountUserCommandHandler.cs

```
@@ -10,27 +12,35 @@ namespace SFA.DAS.EmployerAccounts.ReadStore.Application.Commands
 
-        public CreateAccountUserCommandHandler(IAccountUsersRepository accountUsersRepository)
+        public CreateAccountUserCommandHandler(IAccountUsersRepository accountUsersRepository, ILogger logger)
         {
             _accountUsersRepository = accountUsersRepository;
+            _logger = logger;
         }
 
         public async Task<Unit> Handle(CreateAccountUserCommand request, CancellationToken cancellationToken)
         {
-            var user = await _accountUsersRepository.CreateQuery().SingleOrDefaultAsync(x => x.UserRef == request.UserRef && x.AccountId == request.AccountId, cancellationToken);
-
-            if (user == null)
+            try
             {
-                user = new AccountUser(request.UserRef, request.AccountId, request.Role, request.Created, request.MessageId);
-                await _accountUsersRepository.Add(user, null, cancellationToken);
+                var user = await _accountUsersRepository.CreateQuery().SingleOrDefaultAsync(x => x.UserRef == request.UserRef && x.AccountId == request.AccountId, cancellationToken);
+
+                if (user == null)
+                {
+                    user = new AccountUser(request.UserRef, request.AccountId, request.Role, request.Created, request.MessageId);
+                    await _accountUsersRepository.Add(user, null, cancellationToken);
+                }
+                else
+                {
+                    user.Recreate(request.Role, request.Created, request.MessageId);
+                    await _accountUsersRepository.Update(user, null, cancellationToken);
+                }
             }
-            else
+            catch (UserNotRemovedException exception)
             {
-                user.Recreate(request.Role, request.Created, request.MessageId);
-                await _accountUsersRepository.Update(user, null, cancellationToken);
+                _logger.LogWarning("User has not been removed", exception);
```

Shouldn't the exception be re-thrown rather than swallowed?